defdap/file_readers.py

```diff
@@ -306,15 +309,19 @@ def parseLine(line: str, groupDict: Dict) -> None:
                     round(float(phaseMetadata['gamma']), 3) * np.pi / 180
                 )
             ))
-
+ 
         # Deal with EDX data
+        edx_fields = {}
         if 'EDX Windows' in metadata:
             edx_dict = metadata['EDX Windows']
             num_edx = int(edx_dict['Count'])
+            self.loadedMetadata['EDX Window Count'] = num_edx
             edx_fields = {}
```

Comment:
Not needed, defined above

Reply:
Have refactored to remove duplication in b9ce103. Let me know if any more changes are required. Have tested on OxfordBinary EBSD data with/without EDX and OxfordText data.